apps/dashboard/app/lib/smart_attributes/attributes/auto_batch_clusters.rb

```diff
@@ -17,11 +17,19 @@ def self.build_auto_batch_clusters(opts = {})
       Attributes::AutoBatchClusters.new('auto_batch_clusters', static_opts)
     end
 
+    def self.cluster_options
+      batch_clusters.map do |cluster|
+        cluster_string = cluster.id.to_s
+        max_cores = cluster.job_adapter.nodes.max { |a, b| a.procs <=> b.procs }.procs
```

Comment:
We should cache this call to `cluster.job_adapter.nodes` because it can likely be expensive to make this CLI call all the time.

Additionally we should probably cache `max { |a, b| a.procs <=> b.procs }.procs` too just because it's going to make several hundred comparisons or more.

Reply:
I don't think it really fits into `account_cache.rb` but seems like we need a `cluster_cache.rb` or similar.